### ISSUE TITLE
4.25.0 error response from daemon: timeout and connection refused

### ISSUE THREAD

**Issue by atli-c** [docker/for-mac#7045]
### Description

Docker Desktops internal DNS seems broken.

I cannot pull images or log in to registry. Trying to pull alpine:latest I get error message:
> Error response from daemon: Get "https://registry-1.docker.io/v2/": proxyconnect tcp: dial tcp: lookup http.docker.internal on 192.168.65.7:53: read udp 192.168.65.6:58831->192.168.65.7:53: i/o timeout

Also cannot pull image from self-hosted registry. Note a different error message though:
> Error response from daemon: Get "https://registry.example.net/v2/": proxyconnect tcp: dial tcp: lookup http.docker.internal on 192.168.65.7:53: read udp 192.168.65.6:55912->192.168.65.7:53: read: connection refused
The real domain is replaced with "registry.example.net" for this bug report.

### Reproduce

Fails with "timeout": `docker pull alpine:latest`

Fails with "connection refused":
- docker pull mcr.microsoft.com/playwright:v1.39.0
- docker pull registry.example.net/demo/example (obviously not our real domain. Running [registry:2.8.2](https://hub.docker.com/_/registry))
- docker login -u demo -p demo registry.example.net

### Expected behavior

Should be able to pull images.

### docker version

```bash
Client:
 Cloud integration: v1.0.35+desktop.5
 Version:           24.0.6
 API version:       1.43
 Go version:        go1.20.7
 Git commit:        ed223bc
 Built:             Mon Sep  4 12:28:49 2023
 OS/Arch:           darwin/amd64
 Context:           default

Server: Docker Desktop 4.25.0 (126437)
 Engine:
  Version:          24.0.6
  API version:      1.43 (minimum version 1.12)
  Go version:       go1.20.7
  Git commit:       1a79695
  Built:            Mon Sep  4 12:32:16 2023
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.22
  GitCommit:        8165feabfdfe38c65b599c4993d227328c231fca
 runc:
  Version:          1.1.8
  GitCommit:        v1.1.8-0-g82f18fe
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### docker info

```bash
Client:
 Version:    24.0.6
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.11.2-desktop.5
    Path:     /Users/atc/.docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.23.0-desktop.1
    Path:     /Users/atc/.docker/cli-plugins/docker-compose
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.0
    Path:     /Users/atc/.docker/cli-plugins/docker-dev
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.20
    Path:     /Users/atc/.docker/cli-plugins/docker-extension
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v0.1.0-beta.9
    Path:     /Users/atc/.docker/cli-plugins/docker-init
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     /Users/atc/.docker/cli-plugins/docker-sbom
  scan: Docker Scan (Docker Inc.)
    Version:  v0.26.0
    Path:     /Users/atc/.docker/cli-plugins/docker-scan
  scout: Docker Scout (Docker Inc.)
    Version:  v1.0.9
    Path:     /Users/atc/.docker/cli-plugins/docker-scout

Server:
 Containers: 0
  Running: 0
  Paused: 0
  Stopped: 0
 Images: 2
 Server Version: 24.0.6
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc io.containerd.runc.v2
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 8165feabfdfe38c65b599c4993d227328c231fca
 runc version: v1.1.8-0-g82f18fe
 init version: de40ad0
 Security Options:
  seccomp
   Profile: unconfined
  cgroupns
 Kernel Version: 6.4.16-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 16
 Total Memory: 23.47GiB
 Name: linuxkit-768418011ba5
 ID: bb7082f2-638d-41dc-903b-e499fd291fb9
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: daemon is not using the default seccomp profile
```

### Diagnostics ID

27DAB359-DE09-476F-8F95-DA6A86D2BBB9/20231027123930

### Additional Info

iMac20,1
`uname -mprsv`
> Darwin 23.0.0 Darwin Kernel Version 23.0.0: Fri Sep 15 14:42:42 PDT 2023; root:xnu-10002.1.13~1/RELEASE_X86_64 x86_64 i386

**Comment by atli-c** [comment]
After a few minutes basic docker commands like `docker ps` and `docker version` started hanging. Restarting Docker Desktop fixed it, at least for a while.

**Comment by rocky-hxh** [comment]
same problem here after upgrade to version/4.25.0, and macOS Sonoma

**Comment by namqdam** [comment]
I have the same issue on macOS Monterey (M1) with version 4.25.0

**Comment by t3easy** [comment]
Had the same problem. After giving at least 512MB swap it's working again.

**Comment by t3easy** [comment]
Seams to be the same as #7044

**Comment by atli-c** [comment]
It does seem to be the same as #7044 . I have given Docker Desktop the minimum amount of swap (512 MB) and now it's working.

**Comment by namqdam** [comment]
> It does seem to be the same as #7044 . I have given Docker Desktop the minimum amount of swap (512 MB) and now it's working.

Same here, no working with 0MB swap, then change to 512MB and it works.

**Comment by Sharom** [comment]
> > It does seem to be the same as #7044 . I have given Docker Desktop the minimum amount of swap (512 MB) and now it's working.
> 
> Same here, no working with 0MB swap, then change to 512MB and it works.

Works for me too. Thank you ;)

**Comment by ByteFlyCoding** [comment]
I have the same issue on macOS Monterey (M1) with version 4.25.0

**Comment by biasedbit** [comment]
FWIW, this is consistently happening to me since I upgraded to v4.25.0 on macOS 14.0 with the following resources:
- Memory: 16GB
- Swap: 2GB (increased from 1GB)
- Disk: 120GB

Use case is `pulumi up` with some 7/8 images being built and pushed in parallel. I'll try adding some more resources ¯\\_(ツ)\_/¯

**Comment by djs55** [comment]
Thanks for the report, there is a definitely an issue when the swap is set to zero. I've created a development build with a fix:
 - [Intel Docker.dmg](https://desktop-stage.docker.com/mac/main/amd64/127420/Docker.dmg)
 - [Apple Silicon Docker.dmg](https://desktop-stage.docker.com/mac/main/arm64/127420/Docker.dmg)

If you're having problems with swap, let me know if that helps.

If your case has swap enabled it's probably a different issue. Could you reproduce, upload a diagnostics and quote the ID?

**Comment by h-rub-ksms** [comment]
As a temporary solution, if you change the default configuration of the swap memory (from 1 to 2 for example), the error is solved.

**Comment by bom-d-van** [comment]
I resolve it by switching to https://orbstack.dev/...

**Comment by lorenrh** [comment]
We've released a new version of Docker Desktop containing a fix for this issue, you can see [the release notes](https://docs.docker.com/desktop/release-notes/)! I'll close this issue for now, but if the issue persists, feel free to open a new issue!
